src/components/SingleArticle.jsx

```diff
@@ -3,32 +3,74 @@ import {useEffect, useState} from 'react'
 import { useParams } from 'react-router-dom';
 import  { fetchArticlesById }  from '../../apis';
 import Comments from './Comments';
+import ThumbUpAltOutlinedIcon from '@mui/icons-material/ThumbUpAltOutlined';
+import ThumbDownAltOutlinedIcon from '@mui/icons-material/ThumbDownAltOutlined';
+import Expander from './Expander';
+import { patchArticle } from '../../apis';
 
 const SingleArticle = () => {
     const {article_id} = useParams()
     const [isLoading, setIsLoading] = useState(true)
     const [article, setArticle] = useState({})
+
     useEffect(() => {
         setIsLoading(true)
         fetchArticlesById(article_id)
         .then(({article}) => {
-            setArticle(article[0])
+            setArticle(article)
             setIsLoading(false)
         })
-    }, [])
-    if(isLoading) return <h1>Loading...</h1>
+    }, []);
+
+    function upVote() {      
+        setArticle((currArticle) => {
+            return currArticle.map((article) => {
+                return {...article, votes: article.votes + 1}
+        })
+    });
+    const incVotes = 1
+    patchArticle(article_id, incVotes).catch(err => console.log(err))
+        
+    }
+
+    function downVote() {
+        setArticle((currArticle) => {
+            return currArticle.map((article) => { 
+                return {...article, votes: article.votes - 1}
+        })
+    }); 
+    const decVotes = -1;
+    patchArticle(article_id, decVotes).catch(err => console.log(err))    
```

Comment:
within the catch block I think we'll have to do something more than catch the error.